modules/widgets/src/popup-widget.tsx

```diff
@@ -0,0 +1,97 @@
+/* global document */
+import {FlyToInterpolator, WebMercatorViewport, _GlobeViewport} from '@deck.gl/core';
+import type {Deck, Viewport, Widget, WidgetPlacement} from '@deck.gl/core';
+import {render} from 'preact';
+
+interface PopupWidgetProps {
+  id: string;
+  /**
+   * View to attach to and interact with. Required when using multiple views.
+   */
+  viewId?: string | null;
+  /**
+   * CSS inline style overrides.
+   */
+  style?: Partial<CSSStyleDeclaration>;
+  /**
+   * Additional CSS class.
+   */
+  className?: string;
+  /** 
+   * Position at which to place popup
+   */
+  position: [number, number];
+  /** 
+   * Text of popup
+   */
+  text: string;
+}
+
+export class PopupWidget implements Widget<PopupWidgetProps> {
+  id = 'popup';
+  props: PopupWidgetProps;
+  viewId?: string | null = null;
+  viewport?: Viewport;
+  deck?: Deck<any>;
+  element?: HTMLDivElement;
+
+  constructor(props: PopupWidgetProps) {
+    this.id = props.id || 'popup';
+    this.viewId = props.viewId || null;
+    props.style = props.style || {};
+    props.position = props.position || [0, 0];
+    props.text = props.text || '';
+    this.props = props;
+  }
+
+  setProps(props: Partial<PopupWidgetProps>) {
+    Object.assign(this.props, props);
+    this.update();
+  }
+
+  onViewportChange(viewport) {
+    this.viewport = viewport;
```

Comment:
A broader question since we have eye balls on widgets.. How should we handle if this has been added to no specific viewport, and there are multiple viewports? This remains an unsolved question in all of our viewport-dependent widgets. [Currently](https://github.com/visgl/deck.gl/blob/1c286d094ede876d2edce71cb4b3b71655c20d19/modules/core/src/lib/widget-manager.ts#L255-L263), only the last viewport in the list will get the popup even if the `position` is visible in multiple viewports.

I wonder if we a widget should be able to return multiple UI elements if there are multiple viewports around L66 in onAdd? Or if there should be some warning or documentation to say "you're using multiple viewports, so you should instantiate a widget-per-viewport and supply `viewId`"